pkg/cloud/services/awsnode/cni.go

```diff
@@ -90,6 +90,7 @@ func (s *Service) ReconcileCNI(ctx context.Context) error {
 
 	if s.scope.SecondaryCidrBlock() == nil {
 		if len(envVars) > 0 {
+			s.scope.Info("adding environment properties to vpc-cni", "cluster-name", s.scope.Name(), "cluster-namespace", s.scope.Namespace())
```

Comment:
So this happens continuously. I'm thinking about changing the loop to detect if there are no changes to be applied and then not do an update to the daemonset. What do you think @richardcase?

Reply:
hmm, something is fishy here. I knew something was off in that loop. :D 

It keeps appending environment properties.

```yaml
 Environment:
      ADDITIONAL_ENI_TAGS:                    {}
      AWS_VPC_CNI_NODE_PORT_SUPPORT:          true
      AWS_VPC_ENI_MTU:                        9001
      AWS_VPC_K8S_CNI_CONFIGURE_RPFILTER:     false
      AWS_VPC_K8S_CNI_EXTERNALSNAT:           false
      AWS_VPC_K8S_CNI_LOGLEVEL:               DEBUG
      AWS_VPC_K8S_CNI_LOG_FILE:               /host/var/log/aws-routed-eni/ipamd.log
      AWS_VPC_K8S_CNI_RANDOMIZESNAT:          prng
      AWS_VPC_K8S_CNI_VETHPREFIX:             eni
      AWS_VPC_K8S_PLUGIN_LOG_FILE:            /var/log/aws-routed-eni/plugin.log
      AWS_VPC_K8S_PLUGIN_LOG_LEVEL:           DEBUG
      DISABLE_INTROSPECTION:                  false
      DISABLE_METRICS:                        false
      DISABLE_NETWORK_RESOURCE_PROVISIONING:  false
      ENABLE_IPv4:                            false
      ENABLE_IPv6:                            true
      ENABLE_POD_ENI:                         false
      ENABLE_PREFIX_DELEGATION:               true
      MY_NODE_NAME:                            (v1:spec.nodeName)
      WARM_ENI_TARGET:                        1
      WARM_PREFIX_TARGET:                     1
      ENABLE_IPV6:                            true
      ENABLE_IPV4:                            false
      ENABLE_IPV4:                            false
      ENABLE_IPV4:                            false
```

Reply:
Eh, the deletion is off because it doesn't reset the index and keeps looping on a slice which doesn't have the correct index any longer since an item has been removed. So it keeps adding the wrong properties.

Reply:
Hmm, this is weird. It does do a `break` once it deleted stuff, but it is still off.